addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/service/NetworkService.java

```diff
@@ -29,159 +31,276 @@
+            executorService.awaitTermination(timeoutInMS + 100, TimeUnit.MILLISECONDS);
+            executorService.shutdown();
+            executorService = null;
+            if (!reachable) {
+                updateListener.updatedDeviceState(-1, PresenceDetectionType.NONE);
             }
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt(); // Reset interrupt flag
+            executorService.shutdown();
+            executorService = null;
         }
     }
 
-    public void stopAutomaticRefresh() {
-        refreshJob.cancel(true);
+    /**
+     * Cancel a running refreshJob. Will not throw if no job is running
+     * or the job is already cancelled.
+     */
+    private void cancelRefreshJob() {
         try {
-            ReceiveDHCPRequestPackets.unregister(InetAddress.getByName(hostname).getHostAddress());
-        } catch (UnknownHostException e) {
+            if (refreshJob != null) {
+                refreshJob.cancel(true);
+            }
+        } catch (Exception ignore) {
         }
     }
 
     /**
-     * Updates one device to a new status
+     * Start/Restart a fixed scheduled runner to update the devices reach-ability state.
+     *
+     * @param scheduledExecutorService A scheduler to run pings periodically.
+     * @param stateChangeListener Your callback listener
      */
-    public double updateDeviceState() throws InvalidConfigurationException {
-        int currentTry = 0;
-        do {
-            boolean success;
-            double pingTime;
-
-            try {
-                pingTime = System.nanoTime();
-                if (!useSystemPing) {
-                    success = Ping.checkVitality(hostname, port, timeout);
-                } else {
-                    success = NetworkUtils.nativePing(hostname, port, timeout);
-                }
-                pingTime = System.nanoTime() - pingTime;
-                if (success) {
-                    logger.debug("established connection [host '{}' port '{}' timeout '{}']",
-                            new Object[] { hostname, port, timeout });
-                    return pingTime / 1000000.0f;
-                }
-            } catch (SocketTimeoutException se) {
-                logger.debug("timed out while connecting to host '{}' port '{}' timeout '{}'",
-                        new Object[] { hostname, port, timeout });
-            } catch (IOException ioe) {
-                logger.debug("couldn't establish network connection [host '{}' port '{}' timeout '{}']",
-                        new Object[] { hostname, port, timeout });
-            } catch (InterruptedException e) {
-                logger.debug("ping program was interrupted");
+    public void startAutomaticRefresh(ScheduledExecutorService scheduledExecutorService) {
+        cancelRefreshJob();
+        refreshJob = scheduledExecutorService.scheduleWithFixedDelay(new Runnable() {
```

Comment:
Kai asked basically the same question (https://github.com/openhab/openhab2-addons/pull/1989#discussion_r122158419) ("Do you really think that messing with the scheduled job is the right approach?")

My response was: "Setting a new timeout for the scheduled job makes sense to me, yes. To my understanding the user expresses the wish that as soon as he knows about the (non-)reachable state of a device, he wants the next check to happen after the given period of time. So if the GUI causes a state refresh earlier in time, the next periodic check can be postponed."